docs/central-upgrade.rst

```diff
@@ -25,65 +26,351 @@ Upgrade notes
+  If you run into problems with this step, try stopping Central (``docker compose stop``) and then retry ``docker compose build``.
 
-5. Perform maintenance
+5. **Clean up unused Docker images**
 
-  .. code-block:: console
+.. code-block:: console
 
-    docker prune
+  $ docker image prune
 
 You'll be asked to confirm the removal of all dangling images. Agree by typing the letter ``y`` and pressing ``Enter``.
 
-.. note::
-  If it thinks ``prune`` is not a docker command, run ``docker image prune`` instead.
-
 6. Restart the server
 
-  .. code-block:: console
+.. code-block:: console
+
+  $ docker compose up -d
+
+.. _version-specific-instructions:
+
+Version-specific upgrade instructions
+--------------------------------------
+
+.. _central-upgrade-2023.2:
+
+Upgrading to Central v2023.2
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This is *critical infrastructure upgrade*. It updates many aspects of the infrastructure including upgrading the included PostgreSQL database from 9.6 (no longer supported) to 14 (stable and supported through 2026). It also changes the way that configuration customizations are made to reduce conflicts with future updates to Central. Please read the following instructions carefully.
+
+.. warning::
+  This upgrade may take more time and disk space than previous updates.
+
+.. warning::
+  Before starting:
+
+  #. :doc:`Back up your server <central-backup>`.
+  #. Make sure you have some time available in case something goes wrong (we recommend at least 3 hours). You may want to announce a maintenance window.
+  #. Review upgrade notes for all versions between your current version and the version you are upgrading to.
+
+#. **Log into the web interface** and make a quick note of some of the data you see, such as submission counts and latest submission dates. You may want to use this information to do a quick spot check after the upgrade is finished.
+
+#. **Log into your server.** If you used our :doc:`DigitalOcean installation steps <central-install-digital-ocean>`, we suggest reviewing the section :ref:`central-install-digital-ocean-build` as a reminder, or if you can't remember your password to start at the top of that section to reset your password.
+
+#. **Go to the central folder.**
+
+   .. code-block:: console
+
+     $ cd central
+
+#. **Upgrade Docker if needed.** Check to see if you have Docker Engine v23.x and Docker Compose v2.16.x or greater:
+
+   .. code-block:: console
+
+     $ docker --version && docker compose version
+
+   If you are using old versions, follow the instructions at https://docs.docker.com/engine/install/ubuntu/ to upgrade.
+
+#. **Remove docker-compose.** You will be using ``docker compose`` from now on.
+
+   .. code-block:: console
+
+     $ rm -f `which docker-compose`
+
+#. **Migrate configuration customizations.** This will simplify future Central upgrades. First, check what files have been customized:
+
+   .. code-block:: console
+
+     $ git status
```

Comment:
Was just wondering what some of these instructions would look like if they did the `git pull` step first. How different would cleaning up the config files be? In other words, how important is it that you migrate the custom config options *before* pulling the new code changes?

Reply:
I have a hard time making sense of how to fix the merge conflicts myself, even after doing it a few times. I was not able to come up with a good way to explain it.

Reply:
To be clear, the process I've documented is what I ended up finding the easiest for myself.

Reply:
They way I finally wrapped my head around this section:

1. Make backups for files/service/config.json.template and docker-compose.yml
2. Reset the git contents of those files
3. Things from those two files specifically are moving to `.env` (i'd prefer if i could put them back into .env after i git pull)
4. git stash any other changes you have so you can apply those changes after the upgrade

Reply:
Cool! That looks about like what we have now but we've broken it down a bit more? Most folks will only have changes to either `files/service/config.json.template` or  `docker-compose.yml`, hence the separation. I think copying to the local machine is much easier because of having multiple windows at once.

> i'd prefer if i could put them back into .env after i git pull

Why?